include/aws/cal/private/rsa.h

```diff
@@ -0,0 +1,115 @@
+#ifndef AWS_C_CAL_PRIVATE_RSA_H
+#define AWS_C_CAL_PRIVATE_RSA_H
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/cal/rsa.h>
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/ref_count.h>
+
+struct aws_rsa_key_pair;
+struct aws_der_decoder;
+
+struct aws_rsa_key_vtable {
+    void (*destroy)(struct aws_rsa_key_pair *key_pair);
+
+    int (*encrypt)(
+        struct aws_rsa_key_pair *key_pair,
+        enum aws_rsa_encryption_algorithm algorithm,
+        struct aws_byte_cursor plaintext,
+        struct aws_byte_buf *out);
+    int (*decrypt)(
+        struct aws_rsa_key_pair *key_pair,
+        enum aws_rsa_encryption_algorithm algorithm,
+        struct aws_byte_cursor ciphertext,
+        struct aws_byte_buf *out);
+
+    int (*sign)(
+        struct aws_rsa_key_pair *key_pair,
+        enum aws_rsa_signing_algorithm algorithm,
+        struct aws_byte_cursor digest,
+        struct aws_byte_buf *out);
+
+    int (*verify)(
+        struct aws_rsa_key_pair *key_pair,
+        enum aws_rsa_signing_algorithm algorithm,
+        struct aws_byte_cursor digest,
+        struct aws_byte_cursor signature);
+};
+
+struct aws_rsa_key_pair {
+    struct aws_allocator *allocator;
+    struct aws_rsa_key_vtable *vtable;
+    struct aws_ref_count ref_count;
+
+    size_t key_size_in_bits;
+    struct aws_byte_buf priv;
+    struct aws_byte_buf pub;
+
+    bool good;
```

Comment:
Can we get rid of this? 
ecc doesn't have it
I don't think we have any functions that let the user get their hands on a non-good struct

```suggestion
```